pkg/netbox/query.go

```diff
@@ -0,0 +1,211 @@
+package netbox
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/netbox-community/go-netbox/v4"
+	"github.com/sapcc/concourse-netbox-resource/pkg/concourse"
+	"github.com/sapcc/concourse-netbox-resource/pkg/filter"
+)
+
+func Query(input concourse.Input, ctx context.Context) ([]concourse.Version, error) {
+	var (
+		deviceList      []netbox.DeviceWithConfigContext
+		interfaceList   []netbox.Interface
+		client          *netbox.APIClient
+		err             error
+		netboxFilter    filter.NetboxObject
+		output          []concourse.Version
+		referenceTime   time.Time
+		lastUpdatedTime *time.Time
+	)
```

Comment:
Why are variables declared upfront instead of inlining the declaration with `:=`?

Reply:
To have the variable definitions in one place